scripts/scr_battle_roster/scr_battle_roster.gml

```diff
@@ -121,87 +131,87 @@ function scr_battle_roster(_battle_loci, _loci_specific, _is_planet) {
 
 	                    if (obj_ncombat.battle_special="space_hulk") then obj_ncombat.player_starting_dudes+=1;
 
-	                    if (obj_ini.role[cooh,va]=obj_ini.role[100,12]){col=obj_controller.bat_scout_column;obj_ncombat.scouts+=1;}
-	                    if (obj_ini.role[cooh,va]=obj_ini.role[100,8]){col=obj_controller.bat_tactical_column;obj_ncombat.tacticals+=1;}
-	                    if (obj_ini.role[cooh,va]=obj_ini.role[100,3]){col=obj_controller.bat_veteran_column;obj_ncombat.veterans+=1;}
-	                    if (obj_ini.role[cooh,va]=obj_ini.role[100,9]){col=obj_controller.bat_devastator_column;obj_ncombat.devastators+=1;}
-	                    if (obj_ini.role[cooh,va]=obj_ini.role[100,10]){col=obj_controller.bat_assault_column;obj_ncombat.assaults+=1;}
-	                    if (obj_ini.role[cooh,va]=obj_ini.role[100,17]){col=obj_controller.bat_librarian_column;obj_ncombat.librarians+=1;moov=1;}
-	                    if (obj_ini.role[cooh,va]=string(obj_ini.role[100,17])+" Aspirant"){col=obj_controller.bat_librarian_column;obj_ncombat.librarians+=1;moov=1;}
+	                    if (deploying_unit.role[cooh,va]=deploying_unit.role[100,12]){col=obj_controller.bat_scout_column;obj_ncombat.scouts+=1;}
```

Comment:
is it possible to make this check functional? 
what I mean is, instead of doing an if check for each case, maybe we could make objects/structs compare it internally against a dictionary of stats or something?

Reply:
It could be done, probably. But as long I have not all the combat stuff (mentally) mapped I am somewhat scared of doing any major refactoring or remaking any of it. Even more when obj_ini and obj_controler are involved, since those are used _everywhere_, and if not done correctly will probably break a lot of stuff I'm not even aware of, stuff that will probably not even throw an exception.

But rework it's in order, that's for sure. This stuff it's too confusing and prone to errors to leave it as it is. Will take a look into it.

Reply:
sure, we can get return to this issue later